src/solc_0.8/polygon/child/asset/PolygonAssetERC721.sol

```diff
@@ -35,23 +36,116 @@ contract PolygonAssetERC721 is BaseERC721 {
         address to,
         uint256 id,
         bytes calldata data
-    ) public override(BaseERC721) onlyRole(MINTER_ROLE) {
+    ) public override(BaseERC721, IPolygonAssetERC721) onlyRole(MINTER_ROLE) {
         BaseERC721.mint(to, id, data);
     }
 
     /// @notice Set the metadatahash for a given token id.
     /// @dev The metadata hash for the ERC721 may need to be manually set or overridden.
     /// @param id The token id.
     /// @param uri The full token URI to be used for the token id.
-    function setTokenURI(uint256 id, string memory uri) external onlyRole(METADATA_ROLE) {
+    function setTokenURI(uint256 id, string memory uri) external override onlyRole(METADATA_ROLE) {
         tokenUris[id] = uri;
     }
 
     /// @notice A distinct Uniform Resource Identifier (URI) for a given asset.
     /// @param id The token to get the uri of.
     /// @return URI The token's URI string.
-    function tokenURI(uint256 id) public view override returns (string memory) {
+    function tokenURI(uint256 id) public view override(BaseERC721, IPolygonAssetERC721) returns (string memory) {
         require(ownerOf(id) != address(0), "ZERO_ADDRESS");
         return tokenUris[id];
     }
+
+    function supportsInterface(bytes4 id) public view override(BaseERC721, IPolygonAssetERC721) returns (bool) {
+        return BaseERC721.supportsInterface(id);
+    }
+
+    function setTrustedForwarder(address trustedForwarder)
+        external
+        override(BaseERC721, IPolygonAssetERC721)
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        _trustedForwarder = trustedForwarder;
+    }
```

Comment:
why not use BaseERC721.setTrustedForwarder ?

Reply:
It worked for some cases but gave the following error in some.
![baseERC721 funtioncallError](https://user-images.githubusercontent.com/91669586/176429358-07f5b7cd-4dce-4497-bd43-8927a3ec8c84.png)
One among them is the above case. I tried in this case again but the issue still persists.